Troubleshoot doc on using gRPC client with HTTP/3 (#23857)
dotnet/AspNetCore.Docs: aspnetcore/grpc/troubleshoot.md

* Configures a channel to use `SubdirectoryHandler`.
* Calls the gRPC service with `SayHelloAsync`. The gRPC call is sent to `https://localhost:5001/MyApp/greet.Greeter/SayHello`.

Alternatively, a client factory can be configured with `SubdirectoryHandler` by using <xref:Microsoft.Extensions.DependencyInjection.HttpClientBuilderExtensions.AddHttpMessageHandler%2A>.

::: moniker range="< aspnetcore-6.0"

## Configure gRPC client to use HTTP/3

> The .NET gRPC client supports HTTP/3 with .NET 6 or later. If the server sends an `alt-svc` response header to the client that indicates the server supports HTTP/3, the client will automatically upgrade its connection to HTTP/3. For information about how to enable HTTP/3 on the server, see [Use HTTP/3 with the ASP.NET Core Kestrel web server](xref:fundamentals/servers/kestrel/http3).

Reviewer: ```suggestion
The .NET gRPC client supports HTTP/3 with .NET 6 or later. If the server sends an `alt-svc` response header to the client that indicates the server supports HTTP/3, the client will automatically upgrade its connection to HTTP/3. For information about how to enable HTTP/3 on the server, see <xref:fundamentals/servers/kestrel/http3>.
```
Shortcut to same link.

> HTTP/3 support is in preview in .NET 6, and needs to be enabled via a configuration flag:

Reviewer: ```suggestion
HTTP/3 support is in preview in .NET 6, and needs to be enabled via a configuration flag in the project file:
```

```xml
<ItemGroup>
  <RuntimeHostConfigurationOption Include="System.Net.SocketsHttpHandler.Http3Support" Value="true" />
</ItemGroup>
```

`System.Net.SocketsHttpHandler.Http3Support` can also be set using [AppContext.SetSwitch](xref:System.AppContext.SetSwitch%2A).

> It is also possible to force a gRPC client to use HTTP/3. Forcing HTTP/3 avoids the overhead of upgrading the request. This can be done using the `Http3Handler` specified below:

Reviewer: Yours version is probably better but I loath "it's also possible"

```suggestion
A <xref:System.Net.Http.DelegatingHandler> can bee used to force a gRPC client to use HTTP/3. Forcing HTTP/3 avoids the overhead of upgrading the request. Force HTTP/3 with code similar to the following:
```
